docs/DeveloperGuide.md

```diff
@@ -1,38 +1,190 @@
+<!-- omit in toc -->
 # Developer Guide
+<!-- omit in toc -->
+## Table of contents
+
+- [General Overview](#general-overview)
+- [Design \& implementation](#design--implementation)
+  - [UI class](#ui-class)
+  - [DataStorage class](#datastorage-class)
+    - [loadChassis() method](#loadchassis-method)
+  - [Parser class](#parser-class)
+  - [PC Class Architecture](#pc-class-architecture)
+  - [Command class](#command-class)
+    - [`list [COMPONENT]` Command (builder mode)](#list-component-command-builder-mode)
+- [Appendix: Requirements](#appendix-requirements)
+  - [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+  - [User Stories](#user-stories)
+  - [Non-Functional Requirements](#non-functional-requirements)
+  - [Glossary](#glossary)
+  - [Instructions for manual testing](#instructions-for-manual-testing)
+
+
+## General Overview
+below is the overall architecture diagram for how BigPP works.
+
+![Architecture Diagram](uml-pictures/ArchitectureDiagram.png)
```

Comment:
Shouldn't the diagrams follow UML standards?

![image](https://user-images.githubusercontent.com/109506859/228425542-523b0601-79cd-48e0-bddc-9df6036fb81d.png)

Reply:
Perhaps if the diagrams were to use these non-standard notation, you guys could put in a legend to tell the readers what each notation mean. It could make it clearer for readers like us!
e.g. what is 'A', 'C', 'E', the red hollow box and green filled circle represent?